apps/tlon-mobile/src/screens/GroupChannelsScreen.tsx

```diff
@@ -31,13 +44,84 @@ export function GroupChannelsScreen({
 
   const contactsQuery = store.useContacts();
 
+  const { leaveGroup, togglePinned } = useGroupContext({
+    groupId: groupParam.id,
+  });
+
+  const navigateToGroupSettings = useCallback(
```

Comment:
Could this stuff go in a context, rather than getting drilled down? `NavigationContext` or `AppDataContext` could both be good candidates. `AppDataContext` already includes the `currentUserId`.

Reply:
Absolutely, let me refactor later today.